src/Forms/Form.DevShop.tsx

```diff
@@ -0,0 +1,322 @@
+import { FC, useState } from 'react';
+import { toast } from 'react-toastify';
+import styled from 'styled-components';
+import isEmail from 'validator/lib/isEmail';
+
+import { Form } from '@this/components/Form';
+import { DevShopFormInputs } from '@this/data/types/devShop';
+import axios from 'axios';
+import Spinner from '../components/Elements/Spinner';
+import { TextArea } from '../components/Form/elements';
+import EmailInput from '../components/Form/elements/EmailInput';
+import TextInput from '../components/Form/elements/Input';
+
+type DevShopFormKeys = 'name' | 'company' | 'email' | 'description';
+
+const defaultFormInputs: DevShopFormInputs = {
+  name: '',
+  email: '',
+  company: '',
+  description: '',
+};
+
+type DevShopFormField = {
+  name: string;
+  label: string;
+  errMessage?: string;
+  required?: boolean;
+};
+
+const formFields: Record<keyof DevShopFormInputs, DevShopFormField> = {
+  name: {
+    name: 'fullName',
+    label: 'Name',
+    errMessage: 'Please enter your name',
+    required: true,
+  },
+  email: {
+    name: 'email',
+    label: 'Email',
+    errMessage: 'Please enter a valid email address',
+    required: true,
+  },
+  company: {
+    name: 'company',
+    label: 'Company',
+  },
+  description: {
+    name: 'description',
+    label: 'Project Description',
+    errMessage: 'Please enter a description of your project',
+    required: true,
+  },
+};
```

Comment:
Why not make the `required` field require? I think it will help the next folks remember to actively make a decision if the form field should be required.
```suggestion
type DevShopFormField = {
  name: string;
  label: string;
  errMessage?: string;
  required: boolean;
};

const formFields: Record<keyof DevShopFormInputs, DevShopFormField> = {
  name: {
    name: 'fullName',
    label: 'Name',
    errMessage: 'Please enter your name',
    required: true,
  },
  email: {
    name: 'email',
    label: 'Email',
    errMessage: 'Please enter a valid email address',
    required: true,
  },
  company: {
    name: 'company',
    label: 'Company',
    required: false
  },
  description: {
    name: 'description',
    label: 'Project Description',
    errMessage: 'Please enter a description of your project',
    required: true,
  },
};
```

Reply:
updated here https://github.com/OperationSpark/operationspark-org-website-2022/pull/206/commits/d30c6815fbe56a257c3ae806cbf5f595c21986a0